test/lit/passes/dae_all-features.wast

```diff
@@ -33,7 +33,8 @@
   ;; CHECK-NEXT:  (local.set $0
   ;; CHECK-NEXT:   (i32.const 1)
   ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:  )
```

Comment:
These nested locations surprise me a little. Was the old parser really emitting an empty block there? I am pretty sure it wasn't.

Reply:
I see. Seems reasonable.

Maybe worth checking if say the unoptimized code size tests in Emscripten change with this, just to be safe,
```
./test/runner other.test*unoptimized*
```
(optimized size should not be able to change as empty blocks get removed).

Reply:
Hmm, don't those tests skip Binaryen entirely because they are run with `-O0`?

Reply:
Also this only affects the text parser for now.

Reply:
Oh right, nevermind my worries then.